docs/virtual_machine_template_guide.adoc

```diff
@@ -0,0 +1,189 @@
+
+----
+$ oc create -f <filename>
+----
+
+== Modifying an Uploaded Template
+You can edit a template that has already been uploaded to your project by using the following command: +
+
+----
+$ oc edit template <template>
+----
+
+Writing Templates
+You can define new VirtualMachine templates to make it easy to recreate all relevant objects (offlineVirtualMachine and PersistentVolumeClaim). The template will define the objects it creates along with some metadata to guide the creation of those objects.
+
+[source,yaml]
+----
+apiVersion: kubevirt.io/v1alpha1
+kind: Template
+metadata:
+  name: RHEL-7 ❶
+objects:
+- apiVersion: kubevirt.io/v1alpha1
+  kind: offlineVirtualMachine ❷
+  metadata:
+    name: rhel-7-${NAME} ❸
+  spec:
+    template:
+      metadata:
+      labels:
+        my: label
+      spec:
+        domain:
+          resources: ❹
+            memory: ${MEMORY} 
+            cpu: ${CPU}
+          devices:
+      disks: ➎
+        name: disk0
+        volumeName: mypvc 
+      volumes: ➏
+         name: mypvc
+         presistentVolumeClaim:
+      claimName: rhel-7-pvc-${NAME}
+- apiVersion: kubevirt.io/v1alpha1
+  kind: presistentVolumeClaim ❼
+  metadata:
+    name: rhel-7-pvc-${NAME} ❽
+labels: ...       
+parameters: ...
+
+----
+ 
+❶ The unique name of the template. +
+
+❷ The OfflineVirtualMachine Object. This object must appear only once in the template. +
+
+❸ OfflineVirtualMachine’s name. +
+
+❹ OfflineVirtualMachine’s resources, currently includes only memory and cpu. +
+   Cpu is the number of cpu cores the Virtual Machine requires. +
+   Memory is the minimal amount the Virtual Machine requires. +
+   These attributes are represented as parameters that will either be filled out by the user Or receive default values. +
+   
+➎ Disk contains a “volumeName” attribute, which links it to a volume +
+
+➏ Specification of a disk’s volume from the disks section. +
+        It contains a persistent volume claim.  (more about link:https://kubernetes.io/docs/concepts/storage/persistent-volumes/[Persistent Volume Claims] ) +
+        
+❼ The PersistenVolumeClaim Object that accompanies the OfflineVirtualMachine Object. +
+
+❽ PersistenVolumeClaim’s name. +
+
+=== Labels
+
+Labels will be presented on every OCP template object in order for ManageIQ to recognize it as a KubeVirt template.
+The labels specified in the template are applied to every object that is generated from the template.
+
+[source,yaml]
+----
+labels:
+  # vm-template label applied to all objects created from this template
+  template: “miq.github.io/kubvirt/vm-template”
+----
+
+There is also the ability to add labels in the template from the command line:
+
+----
+$ oc process -f <filename> -l name=“miq.github.io/kubvirt/vm-template”
+----
+
+=== Parameters
+
+Parameters allow a value to be supplied by the user or generated when the template is instantiated. +
```

Comment:
I see `+` artifacts here.

Reply:
these are new line characters for the adoc , is there a better alternative ?

Reply:
I managed to ignore that it is adoc and not markdown :+1: